app/models/agents/weibo_publish_agent.rb

```diff
@@ -76,6 +88,24 @@ def publish_tweet text
       weibo_client.statuses.update text
     end
 
+    def publish_tweet_with_pic text, pic
+      weibo_client.statuses.upload text, open(pic)
+    end
+
+    def valid_image?(url)
+      begin
+        url = URI.parse(url)
```

Comment:
@cantino 
Oh, the function name must be misleading. As you said, I just valid URL(like DNS unresolved) and the content type that returned. If we try to use a pic URL to an invalid or broken pic, the weibo client we use will deal with it.
Actually I wonder is it a good choice to tweet without pic if we met invalid URL, In my use case, I found it annoying to let it error and re-emit one by one, cause I just want to tweet it with or without a pic.